feat: DateTimeProvider

LarsBauer opened the pull request:
Proposal: add abstraction for DateTime object via provider to enable mocking of date-dependent code in unit tests. See sample for usage in plugins.

Possible fake date provider to be added to test framework:
```csharp
public class FakeDateTimeProvider : IDateTimeProvider
{
    private readonly DateTime _dateTime;

    public FakeDateTimeProvider(DateTime dateTime)
    {
      _dateTime = dateTime;
    }

    public DateTime Now => _dateTime.ToLocalTime();
    public DateTime UtcNow => _dateTime.ToUniversalTime();
    public DateTime Today => _dateTime.Date;
}
```

Usage in unit tests:
```csharp
var myDate = new DateTime(2020, 8, 22);

var plugin = new MyPlugin();
plugin.DateTimeProvider = new FakeDateTimeProvider(myDate);
```

LarsBauer committed:
feat: DateTimeProvider

Moberstein reviewed (commented):
Assuming we are using plugin packages could we make use of the `TimeProvider` that comes with .NET 8 through nuget package `Microsoft.Bcl.TimeProvider` for now?
Even if we don't use it, do we maybe want to use same method signature for easier replacement if we ever see .NET update

Also probably should replace then `IStubClock` in [stub](https://github.com/DIGITALLNature/DigitallStub/blob/2161f950a957689c78f6ea86c6c368a043b85168/src/Digitall.Stub/IStubClock.cs)

PepperPix committed:
feat: switch from custom TimeProvider to Microsoft one

usage of Microsoft.Bcl.TimeProvider and Microsoft.Extensions.TimeProvider.Testing for mocking those

PepperPix commented:
change to Microsoft.Bcl.TimeProvider and Microsoft.Extensions.TimeProvider.Testing for mocking those @Moberstein @LarsBauer

PepperPix committed:
build: set buildflag SuppressTfmSupportBuildWarnings

PepperPix committed:
refactor: remove ununsed interface

PepperPix committed:
refactor: change sample to more readable version
